tests/test_uploader.py

```diff
@@ -73,6 +74,10 @@ def test_upload_count_ok(handler, mocker):
     mocker.patch.object(
         uploader, "get_current_upload_count", mocker.Mock(return_value=3)
     )
+    mock_set_pipeline_status = mocker.Mock(return_value=None)
+    mocker.patch.object(
+        uploader, "set_pipeline_status", mock_set_pipeline_status
+    )
```

Comment:
it bothers me that this mocking setup needs to be repeated in every test, so i'm playing around with some pytest stuff to see if I can work out how we could do it only once in a conftest fixture.